### PR TITLE
feat:[CDS-79656]: Disable API calls and return 301 http status code

### PR DESCRIPTION
**Pull request by Fernando-Dourado**
## Describe your changes
- When the FF `CDS_DISABLE_FIRST_GEN_CD` is enabled all the calls to CG should return HTTP status code 301.
- Sounds like we need some work on FE to provide a better UX, you can check the behavior on the below recorded video.

https://github.com/harness/harness-core/assets/102550474/fcda1288-9cad-47eb-9463-b8529b6bd536

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/53053)

### CODE REVIEW

**Comment by Fernando-Dourado** [comment]
trigger uts

**Review by lasdox — DISMISSED** [review]

**Comment by Fernando-Dourado** [comment]
trigger smartchecks

**Review by rafaelmart1ns — APPROVED** [review]

**Comment by Fernando-Dourado** [comment]
trigger ut2

**Comment by Fernando-Dourado** [comment]
trigger uts

File: 400-rest/src/main/java/software/wings/filter/DisableFirstGenFilter.java [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    if (isEmpty(accountId)) {
+      return;
+    }
+
+    // SKIP EVALUATION FOR SOME RESOURCES
+    if (isDelegateRequest(requestContext, resourceInfo) || isNextGenManagerRequest(resourceInfo)) {
+      return;
+    }
+
+    if (ffService.isEnabled(FeatureName.CDS_DISABLE_FIRST_GEN_CD, accountId)) {
```
Comment (arvind-choudhary-h) [inline 400-rest/src/main/java/software/wings/filter/DisableFirstGenFilter.java:66]: I see all calls will go through this DisableFirstGenFilter. Are the FF calls caches in library - couldn't understand firsthand the library flow.

**Comment by Fernando-Dourado** [comment]
trigger ut4, ut2

**Review by lasdox — APPROVED** [review]

**Comment by lasdox** [comment]
I feel like we need some UI enhancement before making this active

File: 400-rest/src/main/java/software/wings/filter/DisableFirstGenFilter.java (hunk 1 above)
Comment (Fernando-Dourado) [reply 400-rest/src/main/java/software/wings/filter/DisableFirstGenFilter.java:66]: @arvind-choudhary-h You are right, all the calls will go to that filter.

The FF service is the same one used on many parts of the CG code, I need to confirm with someone about the cache behavior.

**Comment by Fernando-Dourado** [comment]
> I feel like we need some UI enhancement before making this active

Yes, it's already planned to happen.

**Comment by Fernando-Dourado** [comment]
trigger review

**Comment by Fernando-Dourado** [comment]
More about that can be found at https://harness.slack.com/archives/CJFLDARK9/p1695141174731739

**Pull request merged**